src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java

```diff
@@ -198,12 +198,27 @@ public CaseDetails saveDefendantResponse(
         String authorisation
     ) {
 
-        CCDCase ccdCase = caseMapper.to(claim);
-        ccdCase.setResponse(responseMapper.to(response));
-        ccdCase.setDefendantEmail(defendantEmail);
-        ccdCase.setRespondedAt(nowInUTC());
+        CCDCase ccdCase = CCDCase.builder()
+            .id(claim.getId())
+            .response(responseMapper.to(response))
+            .defendantEmail(defendantEmail)
+            .respondedAt(nowInUTC())
+            .build();
 
-        return update(authorisation, ccdCase, DEFENCE_SUBMITTED);
+        return update(authorisation, ccdCase, CaseEvent.valueOf(getResponseTypeName(response) + "_RESPONSE_SUBMITTED"));
```

Comment:
Do we need to postfix `_RESPONSE_SUBMITTED` ? I think that is coming from `CaseEvent` e.g: `FULL_ADMISSION_RESPONSE_SUBMITTED`